### PR TITLE
[Merged by Bors] - Deploy default and support custom affinities

### PR DESCRIPTION
**Pull request by razvan**
# Description

Part of https://github.com/stackabletech/issues/issues/323

## Review Checklist

- [ ] Code contains useful comments
- [ ] CRD change approved (or not applicable)
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,12 @@ All notable changes to this project will be documented in this file.
 
 - Generate OLM bundle ([#645])
 
-[#645]: https://github.com/stackabletech/commons-operator/pull/645
+### Changed
```
Comment (sbernauer) [inline CHANGELOG.md:11]: What do you think of placing it below Added?

File: docs/modules/zookeeper/pages/pod_placement.adoc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+= Pod Placement
+
+You can configure Pod placement for HDFS nodes as described in xref:concepts:pod_placement.adoc[].
+
+By default, the operator spreads out Apache Zookeeper Pods across cluster nodes.
```
Comment (sbernauer) [inline docs/modules/zookeeper/pages/pod_placement.adoc:5]:
```suggestion
You can configure the Pod placement of the Zookeeper pods as described in xref:concepts:pod_placement.adoc[].

The default affinities created by the operator are:

1. Co-locate all the Zookeeper Pods (weight 20)
```

File: rust/crd/src/affinity.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        let expected: StackableAffinity =
+            validate(get_affinity("simple-zk", &ZookeeperRole::Server)).unwrap();
+
+        let affinity = zk
+            .merged_config(&ZookeeperRole::Server, &rolegroup_ref)
+            .unwrap()
+            .affinity;
+
+        assert_eq!(affinity, expected);
```
Comment (sbernauer) [inline rust/crd/src/affinity.rs:82]: Thx for adding the tests! I personally don't like this assertion. I would prefer not to get the `expected` from the same code path (in `get_affinity`), but instead check the contents "manually" as done in other PRs, such as https://github.com/stackabletech/kafka-operator/pull/557.
I feels to me like nothing is checked at all here :see_no_evil: 
PS: Same below

**Review by sbernauer — APPROVED** [review]
LGTM % Changelog comment

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**